medusa/cassandra_utils.py

```diff
@@ -640,15 +638,35 @@ def is_ccm_up(args, nodetool_command):
         return False
 
 
-def is_cassandra_up(args, port):
+def is_open(host, port):
+    timeout = 3
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.settimeout(timeout)
+    is_accessible = False
     try:
-        args.append(port)
-        subprocess.check_call(args)
-        # check_call returns if call succeeded (returncode is 0). If not exception is triggered
-        return True
-    except subprocess.CalledProcessError:
-        logging.debug('The node is not up yet')
-        return False
+        s.connect((host, port))
+        s.shutdown(socket.SHUT_RDWR)
+        is_accessible = True
+    except socket.error as e:
+        logging.debug('Port {} close on host {}'.format(port, host), exc_info=e)
+        is_accessible = False
+    finally:
+        s.close()
+        return is_accessible
+
+
+def is_cassandra_up(host, port):
+    host_up = False
+    retry = 5
+    for i in range(retry):
```

Comment:
We are using the retrying module elsewhere in Medusa code. Perhaps we can use it here too?

Reply:
@rzvoncek sounds reasonable. Please would you check the last commit.

I will also squash before merge, once you could review the changes of the last commit. The intent was to make changes clear for you to review.